packages/protocol/contracts/common/ProxyCloneFactory.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.5.13;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+import "./InitializableProxy.sol";
+
+contract ProxyCloneFactory is Ownable {
+  address public proxyAddress;
+  event ProxyCreated(InitializableProxy proxy);
+
+  function() external payable {}
+
+  /**
+   * @notice Sets the address of the implementation to clone.
+   * @param The address of the implementation to clone.
+   */
+  function setProxyAddress(address _proxyAddress) external onlyOwner {
+    proxyAddress = _proxyAddress;
+  }
```

Comment:
ditto
`setCloneDNA` or something less silly would cause less confusion imo

Reply:
do we need to start versioning `Proxy` now too?

Reply:
if `Proxy` has a bug, only 1 contract has to be exploited for all users to be affected
then again, this is true for core contracts

Reply:
It's hard to upgrade `Proxy` once it's been deployed, I'm not sure there's a way around this